rclcpp/test/rclcpp/node_interfaces/test_node_parameters.cpp

```diff
@@ -197,5 +197,124 @@ TEST_F(TestNodeParameters, add_remove_parameters_callback) {
 
   RCLCPP_EXPECT_THROW_EQ(
     node_parameters->remove_on_set_parameters_callback(handle.get()),
-    std::runtime_error("Callback doesn't exist"));
+    std::runtime_error("On set parameter callback doesn't exist"));
+}
+
+TEST_F(TestNodeParameters, add_remove_pre_set_parameters_callback) {
+  rcl_interfaces::msg::ParameterDescriptor param1_descriptor;
+  param1_descriptor.name = "param1";
+  param1_descriptor.type = rcl_interfaces::msg::ParameterType::PARAMETER_DOUBLE;
+  param1_descriptor.read_only = false;
+
+  rcl_interfaces::msg::ParameterDescriptor param2_descriptor;
+  param1_descriptor.name = "param2";
+  param1_descriptor.type = rcl_interfaces::msg::ParameterType::PARAMETER_DOUBLE;
+  param1_descriptor.read_only = false;
+
+  node_parameters->declare_parameter(
+      "param1", rclcpp::ParameterValue(0.0), param1_descriptor, false);
+  node_parameters->declare_parameter(
+      "param2", rclcpp::ParameterValue(0.0), param2_descriptor, false);
```

Comment:
We could also test that declaring a parameter does not trigger the pre-set callbacks by moving these after registering a the first callback and checking that their values are unchanged.

Reply:
Done